presto-main/src/main/java/com/facebook/presto/type/setdigest/SetDigest.java

```diff
@@ -183,6 +183,25 @@ public static double jaccardIndex(SetDigest a, SetDigest b)
         return intersection / (double) sizeOfSmallerSet;
     }
 
+    public static double jaccardIndex(SetDigest a, SetDigest b)
+    {
+        int sizeOfSmallerSet = Math.min(a.minhash.size(), b.minhash.size());
```

Comment:
What do you think of something like the following instead

    public static double jaccardIndex(SetDigest a, SetDigest b)
    {
        SetDigest smaller;
        SetDigest larger;
        if (a.minhash.size() < b.minhash.size())
        {
            smaller = a;
            larger = b;
        }
        else {
            smaller = b;
            larger = a;
        }

        LongSortedSet smallerSet = new LongRBTreeSet(smaller.minhash.keySet());
        LongSortedSet largerSet = new LongRBTreeSet(larger.minhash.keySet());

        int intersection = 0;
        for (long key : smallerSet) {
            if (largerSet.contains(key)) {
                intersection++;
            }
        }
        LongSortedSet union = new LongRBTreeSet(largerSet);
        union.addAll(smallerSet);

        return intersection / (double) union.size();
    }